OpenRA.Mods.Common/TraitsInterfaces.cs

```diff
@@ -519,4 +519,10 @@ public class EditorActorDropdown : EditorActorOption
 			OnChange = onChange;
 		}
 	}
+
+	[RequireExplicitImplementation]
+	public interface IPreventMapSpawn
+	{
+		bool PreventMapSpawn(World world, TypeDictionary initDict);
```

Comment:
This doesn't give implementers any way to know the actor type.  Can we please change the `TypeDictionary` to the `ActorReference` or pass the actor type as an argument?

Reply:
done